deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -885,6 +897,8 @@ def save(self):
     self.rnn_final_states = rnn_final_states
     self.rnn_zero_states = rnn_zero_states
     self.session = session
+    if save_dir is not None:
+      shutil.copytree(self.model_dir, save_dir)
```

Comment:
Should it also set `self.model_dir = save_dir`?

Reply:
I think of when you pass "save_dir" to this arg it is creating a "snapshot" of the model at that point in time.  

If you change self.model_dir = save_dir then if you "fit" again the save_dir version will change, not the current model_dir. 

I think the behavior shown here is the more intuitive behavior.

Reply:
Agree with @lilleswing that not changing `self.model_dir` is more intuitive

Reply:
Make sure to clearly document that behavior then.  It's the opposite of what I would expect.